arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,65 @@
 #include <Smartcar.h>
 
-void setup() {
-  // put your setup code here, to run once:
+ArduinoRuntime arduinoRuntime;
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+const int TRIGGER_PIN           = 6; // D6
+const int ECHO_PIN              = 7; // D7
+const unsigned int MAX_DISTANCE = 300;
+SR04 front(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+SimpleCar car(control);
+int magnitude = 0;                              //This we will need later
+
+void setup()
+{
+    Serial.begin(9600);
+    Serial.setTimeout(200);
+}
+
+void loop()
+{
+  handleInput();
+  delay(35);
+}
+
+void handleInput()
+{
+  Serial.println(front.getDistance());
+  if (Serial.available())                           //If the user enters input in the serial The car must move according to that input.
+  {
+   String input = Serial.readStringUntil('\n');
+   if (input.startsWith("m"))     
+   {
+     int cSpeed = input.substring(1).toInt();
+     magnitude = cSpeed;                           //We save the user's input in here in order to have it outside of the if scope.                     
+     car.setSpeed(cSpeed); 
+   } else if (input.startsWith("t"))
+   {
+     int cAngle = input.substring(1).toInt();
+     car.setAngle(cAngle);
+     delay(600);                                                      //This delay is needed for the car to turn in a short while and then go back to its straight direction,  
+   }                                                                  // because we dont want the car to to turn around itself for no reason!
+   if (front.getDistance() > 1 && front.getDistance() < 200)          //This is the obstacle treshhold. The car must detect obstacle closer than 200
```

Comment:
Each call to getDistance is a NEW measurement. This means that if you call it twice, there may very well be different distances being returned. Can you see how that will be the cause of problems?
How would you fix it?

Reply:
I stored the distance.getFront() in a variable and used that variable in the rest of the code to avoid the problem you mentioned. Although I think the previous version was more accurate since it was measuring the distance between car and the obstacle in each instant and since the car is moving, every moment is valuable. Of course I haven't thought about it before you mention the problem, but am I thinking correctly?

Reply:
Yes, your thought is correct in regards to using a single variable for both.
I wouldn't say it's more accurate. If you want to measure if a number is between a particular range (larger than 0 smaller than something) then you should be using the same number, not two different ones 👍